python/ray/data/_internal/numpy_support.py

```diff
@@ -38,6 +39,28 @@ def validate_numpy_batch(batch: Union[Dict[str, np.ndarray], Dict[str, list]]) -
             f"a numpy batch to a block, got: {type(batch)} "
             f"({_truncated_repr(batch)})"
         )
+ 
+def _detect_highest_datetime_precision_dtype(datetime_list: List[datetime]) -> str:
+    highest_precision = 'datetime64[D]'  # Start with day precision
+    
+    for dt in datetime_list:
+        if dt.microsecond != 0:
+            highest_precision = 'datetime64[ns]'
+            break
+        elif dt.second != 0:
+            highest_precision = 'datetime64[s]'
+        elif dt.minute != 0:
+            highest_precision = 'datetime64[m]'
+        elif dt.hour != 0:
+            highest_precision = 'datetime64[h]'
+    
+    return highest_precision
+
+def _convert_datetime_list_to_array(datetime_list: List[datetime]) -> np.ndarray:
+    # Detect highest precision
```

Comment:
Nit: This comment seems superfluous since the function is already name `detect_highest_dataetime_precision`